src/main/java/org/broadinstitute/hellbender/tools/spark/sv/discovery/prototype/InternalFilterLongReadAlignmentsSAMSpark.java

```diff
@@ -0,0 +1,328 @@
+     * Here it is simply appended with a collection operation that collects the alignments stored in the {@link ChimericAlignment}'s.
+     */
+    private static JavaPairRDD<String, List<String>> oldWayOfFiltering(final JavaRDD<GATKRead> longReads,
+                                                                       final SAMFileHeader header,
+                                                                       final Logger toolLogger) {
+
+        // parse SAM records transform to AlignmentInterval format and split gapped alignment
+        final JavaRDD<AlignedContig> parsedContigAlignmentsWithGapSplit
+                = new SAMFormattedContigAlignmentParser(longReads, header, true, toolLogger).getAlignedContigs()
+                .filter(contig -> contig.alignmentIntervals.size()>1).cache();
+
+        // delegates to ChimericAlignment.parseOneContig()
+        return
+                parsedContigAlignmentsWithGapSplit
+                .mapToPair(alignedContig ->
+                        new Tuple2<>(alignedContig.contigName,
+                                ChimericAlignment.parseOneContig(alignedContig, DEFAULT_MIN_ALIGNMENT_LENGTH).stream()
+                                        .flatMap(chimericAlignment -> chimericAlignment.getAlignmentIntervals().stream())
+                                        .sorted(AlignedContig.sortAlignments())
+                                        .collect(Collectors.toList())))
+                .sortByKey()
+                .mapValues(ailist -> ailist.stream().map(AlignmentInterval::toPackedString)
+                                     .collect(SVUtils.arrayListCollector(ailist.size())));
+    }
+
+    /**
+     * Filters an input of SAM file containing alignments of single-ended long read that
+     * aims at providing an "optimal coverage" of the long read, based on an heuristic scoring scheme
+     * {@link #computeScoreOfConfiguration(List, Set, int)}.
+     *
+     * @param longReads     long read alignments
+     * @param header        header for the long reads
+     * @param toolLogger    logger for, most likely, debugging uses
+     *
+     * @return              contigs with alignments filtered and custom formatted as {@link AlignmentInterval}
+     */
+    static JavaRDD<AlignedContig> filterByScore(final JavaRDD<GATKRead> longReads,
+                                                final SAMFileHeader header,
+                                                final Logger toolLogger) {
+
+        toolLogger.info( "Processing these many raw alignments: " + longReads.count() );
+
+        final JavaRDD<AlignedContig> parsedContigAlignments
+                = new SAMFormattedContigAlignmentParser(longReads, header, false, toolLogger)
+                .getAlignedContigs()
+                .filter(InternalFilterLongReadAlignmentsSAMSpark::contigFilter);
+
+        toolLogger.info( "Primitive filtering based purely on MQ left these many contigs: " + parsedContigAlignments.count() );
```

Comment:
it runs in less than a minute on my laptop for our typical assembly.sam, so shouldn't be a big problem for a normal user either. 
But cached as suggested, because the memory requirement is not that high either.